lib/matplotlib/table.py

```diff
@@ -744,6 +746,13 @@ def table(ax,
         cols = len(cellColours[0])
         cellText = [[''] * cols] * rows
 
+    # Check if we have a Pandas DataFrame
+    if _is_pandas_dataframe(cellText):
```

Comment:
Note that `polars.DataFrame.columns` is of type `list[str]`, so the current implementation using `cellText.columns.to_numpy()` would fail. These are exactly the subtleties I did not want to go into as part of this first-time contributor PR. Let's start with pandas, which is a clean addition. We can always generalize later.